### ISSUE TITLE
The usage of '?customerId=' endpoint is deprecated

### ISSUE THREAD

**Issue by raju-cimpress** [commercetools/commercetools-php-sdk#584]
When using CartQueryRequest::of() and putting customerId=xxxxxxx along with other conditions in where clause results in below logs in my project.
```
Call \"in-store/key=xxxxxxxx/carts?customerId=xxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxx\" with method \"GET\" is deprecated: \"The usage of '?customerId=' endpoint is deprecated. Please use '/carts/customer-id=' instead.\"
```
I tried using CartByCustomerIdGetRequest::ofCustomerId() to resolve this deprecation, but it does not allow adding additional where clause to it.
However I noticed that CartByCustomerIdGetRequest used CustomerIdTrait which in end added customerId parameter itself.

In our project customer can have multiple carts based on currency and country.

So could you please update the class CartQueryRequest and add static function ofCustomerId() to create endpoint `/carts/customer-id` and allow additional where conditions on it.

**Comment by jenschude** [comment]
Cause these endpoints don‘t support query parameters. See https://docs.commercetools.com/http-api-projects-carts#get-a-cart-in-a-store-by-customer-id-beta

If you want to query you should use the query endpoints.

**Comment by raju-cimpress** [comment]
Ok. Than how is SDK going to handle deprecation of customerId in query request?

For now I am think of adding  below new class to handle the deprecation, but I would prefer SDK handling the deprecation issues rather than we writing custom class.

```
class CartByCustomerIdQueryRequest extends AbstractQueryRequest
{
    use InStoreTrait;

    protected $resultClass = CartCollection::class;

    /**
     * @param $customerId
     * @param Context|null $context
     */
    public function __construct($customerId, Context $context = null)
    {
        $endPoint = new JsonEndpoint(CartsEndpoint::endpoint() . "/customer-id=$customerId");
        parent::__construct($endPoint, $context);
    }

    public static function ofCustomerId($customerId, Context $context = null)
    {
        return new static($customerId, $context);
    }
}
```

**Comment by jenschude** [comment]
Oh so it seems that we missed to adjust the CartByCustomerIdRequest to avoid the deprecation message. This could easily be fixed.

Besides that if you want to have multiple carts for one customer you have to use the CartsQueryRequest else you are not able to retrieve them as the "Get Cart by Customer ID" endpoint returns always only the last active cart of this customer and doesn't take additional parameters into account.

With the CartsQueryRequest this could look like this:

```
CartQueryRequest::of()
	->where('customerId = :customerId', ['customerId' => '$customerId'])
    ->where('cartState = "' . CartState::ACTIVE . '"')
;
```

**Comment by jenschude** [comment]
I released the v2.11.1 which fixed the deprecation warning sent by the backend

**Comment by raju-cimpress** [comment]
Thanks for the quick release
